src/com/mendix/widget/badge/components/Badge.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, DOM, createElement } from "react";
+import * as classNames from "classnames";
+
+export type BadgeOnclick = "doNothing" | "showPage" | "callMicroflow";
+export type PageSettings = "content" | "popup" | "modal";
+
+export interface MicroflowProps {
+    microflow: string;
+    guid: string;
+}
+
+export interface PageProps {
+    page: string;
+    pageSetting: PageSettings;
+    entity: string;
+    guid: string;
+}
+
+export interface OnClickProps {
+    onClickType: BadgeOnclick;
+    microflowProps?: MicroflowProps;
+    pageProps?: PageProps;
+}
+
+export interface BadgeProps {
+    label?: string;
+    badgeType: string;
+    badgeValue?: string;
+    style?: string;
+    badgeOnClick?: OnClickProps;
+    disabled?: string;
+}
+
+export class Badge extends Component<BadgeProps, {}> {
+    static defaultProps: BadgeProps = {
+        badgeType: "badge",
+        badgeValue: "0",
+        label: "default",
+        style: "default"
+    };
+    componentDidMount() {
+        this.checkConfig();
+        this.setBadge();
+    }
+
+    componentDidUpdate() {
+        this.setBadge();
+    }
+
+    render() {
+        return this.setBadge();
+    }
+
+    private setBadge() {
+        if (this.props.badgeType === "button") {
+            return this.createBadgeButton();
+        } else if (this.props.badgeType === "label") {
+            return this.createBadgeLabel();
+        } else {
+            return this.createBadge();
+        }
+    }
+
+    private createBadge() {
+        return createElement("div",
+            {
+                className: classNames("widget-badge-display",
+                    { "widget-badge-link": !!this.props.badgeOnClick }
+                ),
+                onClick: () => this.handleOnClick(this.props.badgeOnClick)
```

Comment:
This actually allocates memory for and creates a new closure every time render is called. That is not needed, just create it once in the constructor and reuse it.